lib/spack/spack/modules.py

```diff
@@ -79,8 +74,35 @@ def print_help():
             "")
 
 
+def inspect_path(path):
+    class PathInspector(object):
+        dirname2varname = {
+            'bin': ('PATH',),
+            'man': ('MANPATH',),
+            'lib': ('LIBRARY_PATH', 'LD_LIBRARY_PATH'),
+            'lib64': ('LIBRARY_PATH', 'LD_LIBRARY_PATH'),
+            'include': ('CPATH',),
+            'pkgconfig': ('PKG_CONFIG_PATH',)
+        }
+
+        def __call__(self, env, directory, names):
+            for name in names:
+                variables = PathInspector.dirname2varname.get(name, None)
+                if variables is None:
+                    continue
+                absolute_path = join_path(os.path.abspath(directory), name)
+                for variable in variables:
+                    env.prepend_path(variable, absolute_path)
+
+    env, inspector = EnvironmentModifications(), PathInspector()
+    os.path.walk(path, inspector, env)
```

Comment:
@tgamblin Ok, will restore the previous logic. It seems in fact better to leave the responsibility of adding non-standard directories to the package maintainer.